diff --git a/src/security/credentials.rs b/src/security/credentials.rs
index 1627fa24b..234b4e0bf 100644
--- a/src/security/credentials.rs
+++ b/src/security/credentials.rs
@@ -338,7 +338,7 @@ impl ChannelCredentials {
         unsafe {
             grpc_sys::grpc_init();
         }
-        let creds = unsafe { grpc_sys::grpc_google_default_credentials_create(ptr::null_mut()) };
+        let creds = unsafe { grpc_sys::grpc_google_default_credentials_create(ptr::null_mut(), ptr::null_mut()) };
         if creds.is_null() {
             Err(Error::GoogleAuthenticationFailed)
         } else {
